src/fabric.c

```diff
@@ -62,12 +62,7 @@ pthread_mutex_t ini_lock = PTHREAD_MUTEX_INITIALIZER;
 
 static struct fi_filter prov_filter;
 
-struct fi_provider core_prov = {
-	.name = "core",
-	.version = 1,
-	.fi_version = FI_VERSION(FI_MAJOR_VERSION, FI_MINOR_VERSION)
-};
-
+extern struct fi_provider core_prov;
```

Comment:
Just curious: why is this an extern in three `.c` files instead of a single `.h` file somewhere?

```sh
$ ack 'extern struct fi_provider core_prov'
src/log.c
64:extern struct fi_provider core_prov;

src/unix/osd.c
47:extern struct fi_provider core_prov;

src/var.c
47:extern struct fi_provider core_prov;
```